template/src/ontology/run.sh.jinja2

```diff
@@ -23,10 +30,20 @@ if [ x$ODK_DEBUG = xyes ]; then
     TIMECMD="/usr/bin/time -f ### DEBUG STATS ###\nElapsed time: %E\nPeak memory: %M kb"
 fi
 
-docker run -v $PWD/../../:/work -w /work/src/ontology -e ROBOT_JAVA_ARGS="$ODK_JAVA_OPTS" -e JAVA_OPTS="$ODK_JAVA_OPTS" {% if project.use_external_date is sameas True %}-e TODAY=`date +%Y-%m-%d` {% endif %}--rm -ti obolibrary/$IMAGE $TIMECMD "$@"
+VOLUME_BIND=$PWD/../../:/work
+WORK_DIR=/work/src/ontology
+
+if [ -n "$USE_SINGULARITY" ]; then
+    export ROBOT_JAVA_ARGS="$ODK_JAVA_OPTS"
+    export JAVA_OPTS="$ODK_JAVA_OPTS"
+    {% if project.use_external_date is sameas True %}export TODAY=$(date +%Y-%m-%d){% endif %}
+    singularity exec --bind $VOLUME_BIND -W $WORK_DIR docker://obolibrary/$IMAGE:$TAG $TIMECMD "$@"
+else
+    docker run -v $VOLUME_BIND -w $WORK_DIR -e ROBOT_JAVA_ARGS="$ODK_JAVA_OPTS" -e JAVA_OPTS="$ODK_JAVA_OPTS" {% if project.use_external_date is sameas True %}-e TODAY=`date +%Y-%m-%d` {% endif %}--rm -ti obolibrary/$IMAGE $TIMECMD "$@"
```

Comment:
Its odd that we do not use the $TAG variable for the docker run command - There is a tiny risk that a client is using:

```
IMAGE=odklite:1.2.3 ./run.sh make ...
```

Which will now cause the script to fail. 

Is adding the `$TAG` variable strictly necessary for the singularity case above?

Reply:
Yes, singularity won't assume latest implicitly. We could add some checks to see if image already includes a TAG and we could then set TAG from there.

Reply:
Basically like: `if ':' not in IMAGE`, IMAGE=IMAGE+TAG`?

Reply:
Addressed in [d5d5513](https://github.com/INCATools/ontology-development-kit/pull/640/commits/d5d5513920b74e9729e7c3f173b9064f480d1621).